File: easy/morse_code.c
#include <stdio.h>

int main(int argc, char *argv[])
{
	FILE *fp;
	char c;
	int m = 1;
	char *morse = "ETIANMSURWDKGOHVF L PJBXCYZQ  54 3   2       16       7   8 90";
	fp = fopen(*++argv, "r");
	while ((c = getc(fp)) != EOF) {
		if (c == '.') {
			m <<= 1;
		} else if (c == '-') {
			m = (m << 1) + 1;
		} else if (m == 1) {
			putchar(' ');
		} else {
			if (m < 64)
				putchar(morse[m-2]);
			if (c == '\n')
				putchar('\n');
			m = 1;
		}
	}
	return 0;
}

File: easy/morse_code.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
)

const morse = "ETIANMSURWDKGOHVF L PJBXCYZQ  54 3   2       16       7   8 90"

func main() {
	data, err := os.Open(os.Args[1])
	if err != nil {
		log.Fatal(err)
	}
	defer data.Close()
	scanner := bufio.NewScanner(data)
	scanner.Split(bufio.ScanBytes)
	m := 1
	for scanner.Scan() {
		c := rune(scanner.Text()[0])
		switch {
		case c == '.':
			m <<= 1
		case c == '-':
			m = (m << 1) + 1
		case m == 1:
			fmt.Print(" ")
		default:
			if m < 64 {
				fmt.Print(string(morse[m-2]))
			}
			if c == '\n' {
				fmt.Println()
			}
			m = 1
		}
	}
	if m > 1 {
		if m < 64 {
			fmt.Print(string(morse[m-2]))
		}
		fmt.Println()
	}
}
